reconcile/checkpoint.py

```diff
@@ -30,6 +30,7 @@
 )
 
 
+@lru_cache
```

Comment:
We've had a lenghty discussion elsewhere about `lru_cache` leaking memory. Maybe we should prefer `cache`?

Reply:
wasn't it related to the use of `self`?